samples/nrf9160/nrf_cloud_coap_client/src/main.c

```diff
@@ -0,0 +1,803 @@
+		gnss.pvt.lon = -122.685978;
+		gnss.pvt.accuracy = 30;
+	}
+
+	printk("\n***********************************************\n");
+	switch (cur_test) {
+	case 1:
+		printk("**** %d. Getting pending FOTA job execution ****\n", cur_test);
+		err = handle_fota_process();
+		if (err != -EAGAIN) {
+			LOG_INF("FOTA check completed.");
+		}
+		break;
+	case 2:
+		printk("*** %d. Sending temperature ********************\n", cur_test);
+		err = nrf_cloud_coap_sensor_send(NRF_CLOUD_JSON_APPID_VAL_TEMP, temp);
+		if (err) {
+			LOG_ERR("Error sending sensor data: %d", err);
+			break;
+		}
+		LOG_INF("Sent %.1f C", temp);
+		temp += 0.1;
+		break;
+	case 3:
+		printk("*** %d. Getting position ***********************\n", cur_test);
+#if defined(CONFIG_WIFI)
+		err = scan_wifi_start(&wifi_scan_sem);
+		LOG_INF("Waiting for Wi-Fi scans...");
+		if (err) {
+			LOG_ERR("Error starting Wi-Fi scan: %d", err);
+			err = 0;
+			break;
+		}
+		k_sem_take(&wifi_scan_sem, K_SECONDS(APP_WAIT_WIFI_S));
+		wifi_info = scan_wifi_results_get();
+#endif
+		LOG_INF("Waiting for neighbor cells..");
+		err = k_sem_take(&cell_info_ready_sem, K_SECONDS(APP_WAIT_CELLS_S));
+		if (err) {
+			LOG_ERR("Timeout waiting for cells: %d", err);
+			err = 0;
+			if (!IS_ENABLED(CONFIG_WIFI)) {
+				break;
+			}
+		}
+
+		(void)k_mutex_lock(&cell_info_mutex, K_FOREVER);
+
+		if (cell_info.current_cell.id != LTE_LC_CELL_EUTRAN_ID_INVALID) {
+			LOG_INF("Current cell info: Cell ID: %u, TAC: %u, MCC: %d, MNC: %d",
+				cell_info.current_cell.id, cell_info.current_cell.tac,
+				cell_info.current_cell.mcc, cell_info.current_cell.mnc);
+		} else {
+			LOG_WRN("No current serving cell available");
+		}
+
+		if (cell_info.ncells_count || cell_info.gci_cells_count) {
+			LOG_INF("Performing multi-cell request with "
+				"%u neighbor cells and %u GCI cells",
+				cell_info.ncells_count, cell_info.gci_cells_count);
+		} else {
+			LOG_INF("Performing single-cell request");
+		}
+
+		loc_request.cell_info = &cell_info;
+		loc_request.wifi_info = wifi_info;
+		err = nrf_cloud_coap_location_get(&loc_request, &result);
+		(void)k_mutex_unlock(&cell_info_mutex);
+		if (err) {
+			LOG_ERR("Unable to get location: %d", err);
+			break;
+		}
+
+		if ((result.type >= 0) && (result.type < LOCATION_TYPE__INVALID)) {
+			type = location_types[result.type];
+		} else {
+			type = "unknown";
+		}
+		/* Process the returned location once it arrives */
+		gnss.pvt.lat = result.lat;
+		gnss.pvt.lon = result.lon;
+		gnss.pvt.accuracy = result.unc;
+		LOG_INF("Location: %s, %.12g, %.12g, %d", type,
+			result.lat, result.lon, result.unc);
+
+		request_cells = true;
+		break;
+	case 4:
+		printk("*** %d. Sending GNSS PVT ***********************\n", cur_test);
+		err = nrf_cloud_coap_location_send(&gnss);
+		if (err) {
+			LOG_ERR("Error sending GNSS PVT data: %d", err);
+		} else {
+			LOG_INF("PVT sent");
+		}
+		break;
+	case 5:
+		printk("*** %d. Getting A-GPS data *********************\n", cur_test);
```

Comment:
print this only if A-GPS is defined and  #else print something saying it will be skipped?